src/main.c

```diff
@@ -20,6 +21,68 @@ void TVMPlatformAbort(int error) {
   for (;;) ;
 }
 
+u32_t g_utvm_start_time;
+
+#define MILLIS_TIL_EXPIRY 200
+#define TIME_TIL_EXPIRY (K_MSEC(MILLIS_TIL_EXPIRY))
+K_TIMER_DEFINE(g_utvm_timer, /* expiry func */ NULL, /* stop func */ NULL);
+
+int g_utvm_timer_running = 0;
+
+int TVMPlatformTimerStart() {
```

Comment:
just wondering if you tried this approach? https://docs.zephyrproject.org/1.9.0/kernel/timing/clocks.html#measuring-time-with-high-precision

Reply:
actually, ignore me :)

Reply:
one thing i'm unsure of is whether the hybrid method i've implemented makes any sense (i.e., using the kernel timer for high precision on quick executions and falling back to the interrupt-based timer when execution time exceeds the capacity of the high-precision route), or if we think op execution should never go on _that_ long